### ISSUE TITLE
[Question] Where are the standard library objects such as Array or Object?

### ISSUE THREAD

**Issue by sigmasoldi3r** [maziac/binary-file-viewer#14]
Hello! I am trying to use some of the data that I read to interpret the following (IE: Reading a palette to draw an image on a canvas later on).

The thing is that calls like `Object.keys` or `Array(10)` give errors, as `Object` and `Array` are not defined. The same happens with `JSON`.

**Comment by maziac** [comment]
Yes, that's true.
The binary file viewer has restrictions and does not allow for the complete js.

**Comment by sigmasoldi3r** [comment]
oh that's sad to hear... ~~I hope arrays and objects are mutable anyway. (Are they?)~~

Edit:

I'm trying to store data that I'm reading within an object (like `obj[key] = value;`), but the object is always empty :cry: 

How can I circumvent that?

## Edit 2:
(Yes objects are mutable)

I've found that I can do mutations outside `addDetails()`!

**Comment by maziac** [comment]
This here works:
~~~
	let obj = {};
	obj.mykey = 'mykeyvalue';
...
	addRow('WORD 1', obj["mykey"]);
	addRow('WORD 1b', obj.mykey);
~~~

But maybe your use case is too complex.

**Comment by sigmasoldi3r** [comment]
Only if there was a way to make things like base64 encode or read as an arraybuffer... that could do the trick :thinking:

**Comment by maziac** [comment]
In scopelessfunctioncall.ts you find the function
~~~js
import * as vm from 'vm';
...
export function vmRunInNewContext(funcBodyString: string, parameters = {}, filePath?: string): any {
	//	console.time('vmRunInNewContext');
	const result = vm.runInNewContext(funcBodyString, parameters, filePath);
	//	console.timeEnd('vmRunInNewContext');
	return result;
}
~~~

It executes "your" parser js file.
Everything that the vm supports is supported, but nothing else.

**Comment by sigmasoldi3r** [comment]
And classes like `Object`, `Array` and function like `atob` and `btoa` can't be included? They would help a lot to process complex data.

**Comment by maziac** [comment]
I'm not sure how to do that. 
But you can fork the repo and try yourself.

**Comment by maziac** [comment]
Hi, 
I have been working on this and found a way to add Number, Object, atob and btoa.
Could you please try if this vsix would work for you.
[binary-file-viewer-1.9.0-rc1.vsix.zip](https://github.com/maziac/binary-file-viewer/files/14736841/binary-file-viewer-1.9.0-rc1.vsix.zip)

**Comment by maziac** [comment]
Implemented in v1.9.0: Object, Array, Map, JSON, atob and btoa
